projects/plugins/jetpack/_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -675,6 +675,25 @@ public static function register_endpoints() {
 			)
 		);
 
+		// Return all module settings.
+		register_rest_route(
+			'jetpack/v4',
+			'/licensing/set-license',
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => __CLASS__ . '::set_jetpack_license',
+				'permission_callback' => __CLASS__ . '::view_admin_page_permission_check',
```

Comment:
Should this be accessible to everyone who has access to the admin page? This would include a site's subscribers. Maybe checking for the `jetpack_connect_user` capability would be more appropriate?

Reply:
I've added a custom permission callback `set_jetpack_license_key_permission_check`. It only checks for `jetpack_connect_user`, similar to some other permission callbacks, but I added this as a separate callback so we can change the condition specifically for that route if needed.